Fix #15616 error: [FATAL] Invalid schema changes detected during integrity checks

vnguyen42 opened the pull request:
Hello,

Here is a fix for the issue #15616 which occur when you try to use strapi import on a Windows based system and trigger a fatal error at import.

The error come from the fact the path separator used in the code ('/') is the wrong one, and not the one used by Windows '\'.

Others solutions fixed it on Windows but broke it on other platforms. This PR allows to fix the problem on both Windows while keeping it working on other Unix based system, and in production.

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/strapi?pullRequest=16212) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/strapi?pullRequest=16212) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/strapi?pullRequest=16212) it.</sub>

vnguyen42 commented:
For those who are looking to fix the issue directly in their live project :

**Head to :** 
node_modules/@strapi/data-transfer/lib/file/providers/source/index.js

**At line 144, replace :** 
`const parts = path_1.default.relative('.', filePath).split('/');`

**with :** 
`const parts = path_1.default.relative('.', filePath).split(os_splitter);`

**At line 30, after :** 
`const providers_1 = require("../../../errors/providers");`

**Add :**

```
const os = require('os');
const os_splitter = os.platform() === 'win32' ? '\\' : '/';

```

innerdvations commented:
Thank you for the PR!

I'm closing this one in favor of https://github.com/strapi/strapi/pull/16402 which also fixes the issue of win32-generated archives working on posix systems and a few other related items.

I've attempted to give you co-author credit in a commit which I haven't tried before, hopefully that works :)
